### PR TITLE
Track US States Vaccine Coverage Data #990

### PR DESCRIPTION
**Pull request by comster**
- [x] Scraper
- [x] API endpoints
- [x] Swagger Docs
- [x] Unit tests
- [x] Scraper config

### CODE REVIEW

**Comment by comster** [comment]
@ebwinters how does this look?

**Comment by ebwinters** [comment]
@comster will review today, thanks for the contribution!

File: config/index.js [hunk 1]
```diff
@@ -33,6 +33,8 @@ config.vaccineInterval = process.env.VACCINE_INTERVAL || 864e5;
 // eslint-disable-next-line camelcase
 config.vaccineCoverageInterval = process.env.VACCINE_COVERAGE_INTERVAL || 864e5;
 // eslint-disable-next-line camelcase
+config.vaccineStateCoverageInterval = process.env.VACCINE_STATE_COVERAGE_INTERVAL || 864e5;
```
Comment (ebwinters) [inline config/index.js:36]: if it's from the same source we don't really need a new config for the interval just sayin

File: public/apidocs/swagger_v3.json [hunk 2]
```diff
@@ -1217,6 +1217,65 @@
         }
       }
     },
+    "/v3/covid-19/vaccine/coverage/states": {
```
Comment (ebwinters) [inline public/apidocs/swagger_v3.json:1220]: love that you already wrote docs

File: scrapers/covid-19/getVaccine.js [hunk 3]
```diff
@@ -47,6 +47,16 @@ const groupVaccineDataByCountry = (vaccineData) => {
 	};
 };
 
+const groupVaccineDataByState = (vaccineData) => {
+	const groupedByStateVaccineDataObject = {};
+
+	vaccineData.forEach((timelineData) => {
+		if (groupedByStateVaccineDataObject[timelineData.location] === undefined) groupedByStateVaccineDataObject[timelineData.location] = [timelineData];
```
Comment (ebwinters) [inline scrapers/covid-19/getVaccine.js:54]: nitpick: use braces see https://blog.codecentric.de/en/2014/02/curly-braces/

File: scrapers/covid-19/getVaccine.js [hunk 4]
```diff
@@ -173,6 +183,65 @@ const generateSpecificCountryVaccineData = (timelineData) => {
 	return { country: '', countryInfo: null, timeline: updatedTimeline };
 };
 
+/**
+ * Formats raw timeline data
+ * @param {array} timelineData An array of raw timeline data
+ * @returns {object}
+ */
+
+const generateSpecificStateVaccineData = (timelineData) => {
```
Comment (ebwinters) [inline scrapers/covid-19/getVaccine.js:192]: this method copies a ton of code from the above method, is there any way you can merge them into one generic method to reuse the code?

File: serverScraper.js [hunk 5]
```diff
@@ -23,6 +24,8 @@ setInterval(excecuteScraperGov, config.govInterval);
 setInterval(excecuteScraperVaccine, config.vaccineInterval);
 // Update vaccine data every  24 hours
 setInterval(excecuteScraperVaccineCoverage, config.vaccineCoverageInterval);
+// Update vaccine data every  24 hours
+setInterval(excecuteScraperVaccineStateCoverage, config.vaccineStateCoverageInterval);
```
Comment (ebwinters) [inline serverScraper.js:28]: as I said, can just reuse this config variable here `vaccineCoverageInterval`

**Review by ebwinters — CHANGES_REQUESTED** [review]
honestly great work here just a couple things I think you could make more concise 😃

File: config/index.js (hunk 1 above)
Comment (comster) [reply config/index.js]: fair enough

File: scrapers/covid-19/getVaccine.js (hunk 3 above)
Comment (comster) [reply scrapers/covid-19/getVaccine.js]: done.

File: scrapers/covid-19/getVaccine.js (hunk 4 above)
Comment (comster) [reply scrapers/covid-19/getVaccine.js]: Sure.

File: serverScraper.js (hunk 5 above)
Comment (comster) [reply serverScraper.js]: done.

**Comment by comster** [comment]
Thanks for reviewing this @ebwinters 

Looking forward to having more data available 😄

**Pull request merged**